### PR TITLE
Improve conductor call error messages

### PR DESCRIPTION
**Pull request by timotree3**
Previously whenever a zome call failed, all you saw in chaperone was `CONDUCTOR CALL ERROR: [object Object]`. With this PR, you would actually see the error

Update: I ended up updating the mock conductor dependency in order to write tests for the improved error message, it turns out that the old version of mock conductor we were using was not detecting a lot of things that were wrong with the old tests. I fixed all those problems in this PR as well. Let me know if you want me to split it out into two PRs

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
   },
   "devDependencies": {
     "@holo-host/chaperone": "git://github.com/Holo-Host/chaperone.git#620c84181efff68ea30eec8e2c3fdd94692c3921",
-    "@holo-host/mock-conductor": "^0.3.0",
+    "@holo-host/mock-conductor": "^0.2.1",
```
Comment (zo-el) [inline package.json]: why did we move back in the version?

**Review by zo-el — APPROVED** [review]
looks good

File: package.json (hunk 1 above)
Comment (timotree3) [reply package.json]: I just released 0.2.1 yesterday and I don't know where 0.3.0 came from. Prior to my release the master branch had `version: 0.2.0`

Comment (timotree3) [reply package.json]: Just released 0.3.1 to resolve this

**Pull request merged**